tests/api-compatibility/README.md

```diff
@@ -0,0 +1,126 @@
+The task will expect the following parameters:
+
+`ApiCompatPath` - Path where Microsoft.DotNet.ApiCompat nuget package is located.
+
+`ApiLevel` - Api level just built.
+
+`LastStableApiLevel` - The last stable api level.
+
+`TargetImplementationPath` - Path to the implementation assembly. Based on this path we will generate the contract path by replace current version with previous version
+
+`ApiCompatibilityPath` - Path to the Api Compatibility folder. On this folder we should find the acceptable-breakages files and the reference folder that contians the current Mono.Android.dll assembly.
+
+The *task* is aware of the Api levels and what is the previous Api level for agiven version, so When building api version 2, the Task knows if previous version is the version 1.5 or 1.0 for instance.
+
+## Usage
+
+To use the build task we need to import it to our project. On *Mono.Android.csproj* we add the follow line to refrence the build task to our project:
+
+```<UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\Xamarin.Android.Tools.BootstrapTasks.dll" TaskName="Xamarin.Android.Tools.BootstrapTasks.CheckApiCompatibility" />```
+
+Once task is referenced we can use it by adding the follow lines:
+
+```
+  <PropertyGroup>
+    <ApiCompatibilityDir>$(MSBuildProjectDirectory)/Test/api-compatibility</ApiCompatibilityDir>
+  </PropertyGroup>
+  <ItemGroup>
+    <ApiCompatibilityFiles Include="$(ApiCompatibilityDir)/acceptable-breakages-v*.txt" />
+    <ApiCompatibilityFiles Include="$(ApiCompatibilityDir)/reference/*.*" />
+  </ItemGroup>
+  <Target Name="AfterBuild" Inputs="$(TargetPath);@(ApiCompatibilityFiles)" Outputs="$(IntermediateOutputPath)CheckApiCompatibility.stamp">
+    <CheckApiCompatibility
+        ApiCompatPath="../../packages/Microsoft.DotNet.ApiCompat.5.0.0-beta.19562.5/tools/net472/"
+        ApiLevel="$(AndroidFrameworkVersion)"
+        LastStableApiLevel="$(AndroidLatestStableFrameworkVersion)"
+        TargetImplementationPath="$(OutputPath)"
+        ApiCompatibilityPath = "$(ApiCompatibilityDir)" />
+    <Touch Files="$(IntermediateOutputPath)CheckApiCompatibility.stamp" AlwaysCreate="True" />
+    <ItemGroup>
+      <FileWrites Include="$(IntermediateOutputPath)CheckApiCompatibility.stamp" />
+    </ItemGroup>
+  </Target>
+```
+
+`ApiCompatibilityFiles` - Are the files we would like to monitor for changes, this is doen to prevent the task to run if it is not required. (there were no changes)
+
+`Touch Files` - It is our timestamp file to control if task should run.
+
+[bst]: https://github.com/xamarin/xamarin-android/tree/master/build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks
+
+## ApiCompatibilityPath
+
+On this path we will place the files that contain all acceptable breakages for a given Api version.
```

Comment:
These files need to follow some form of "file format".  Are there docs for that file format?  What's a line look like?

An example is provided below, but we don't know what are the available messages, nor -- more importantly! -- what the *workflow* is for updating these files.

*Presumably* some tool will print out these "rules" for us.  What is that tool?  How do we obtain the output?  Where does the output *go* so that new errors are ignored (after review)?

Reply:
The files are the exact content of the output of Microsoft.DotNet.ApiCompat.exe.
The best workflow is to build Mono.Android, and if any breakage happen, see the error printed on the screen or log, and decided if it is an acceptable breakage, just add that exact line to the acceptable-*.txt file